Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift

```diff
@@ -50,34 +50,59 @@ public struct AutomaticCuration {
         withTraits variantsTraits: Set<DocumentationDataVariantsTrait>,
         context: DocumentationContext
     ) throws -> [TaskGroup] {
-        // Get any default implementation relationships for this symbol
-        //
-        // It's okay to include all variants here because we just use this set to filter
-        // out these values from automatic curation.
-        let defaultImplementationReferences = Set<String>(
-            (node.semantic as? Symbol)?.defaultImplementationsVariants.allValues
-                .lazy
-                .map(\.variant)
-                .flatMap(\.implementations)
-                .compactMap { implementation in
-                    implementation.reference.url?.absoluteString
-                } ?? []
-        )
+        let languagesFilter = Set(variantsTraits.compactMap {
+            $0.interfaceLanguage.map { SourceLanguage(id: $0) }
+        })
         
-        return try context.children(of: node.reference)
-            // Remove any default implementations
-            .filter({ (reference, kind) -> Bool in
-                return !defaultImplementationReferences.contains(reference.absoluteString)
-            })
+        let children: [ResolvedTopicReference]
+        if languagesFilter.isEmpty {
+            // If the caller requests a combined topic section for all source languages, then the results from the `TopicGraph` will be accurate.
+            
+            // Get any default implementation relationships for this symbol
+            //
+            // It's okay to include all variants here because we just use this set to filter
+            // out these values from automatic curation.
+            let defaultImplementationReferences = Set<String>(
+                (node.semantic as? Symbol)?.defaultImplementationsVariants.allValues
+                    .lazy
+                    .map(\.variant)
+                    .flatMap(\.implementations)
+                    .compactMap { implementation in
+                        implementation.reference.url?.absoluteString
+                    } ?? []
+            )
+            
+            children = context.children(of: node.reference)
```

Comment:
Similar to directDescendants(of:languagesFilter) - wouldn't this be easier to read and more natural if it was an instance method of node?

It looks like `children(of:)` uses the topic graph, and not the path hierarchy, so maybe that doesn't work. It might be nice to name these methods ("children" or "directDescendants") based on whether they use the topic graph or the path hierarchy.

Reply:
Like in the other case; since `node` is a `DocumentationNode` moving the API to the node would only invert the declaration:

`Context . method (Node,    Languages)` vs
`Node    . method (Context, Languages)`

---

In [this follow up PR](https://github.com/apple/swift-docc/pull/856) I stop using the topic graph (`children(of:)`) for automatic curation _completely_, so this code will only see one of those names. This new function is only internal so we can rename it or change it without causing breakage changes.